src/files.ts

```diff
@@ -248,29 +254,24 @@ export async function* collectBundleFiles({
   }
 
   // Scan folders
-  const globPatterns = getGlobPatterns(options.supportedFiles);
+  const globPatterns = getGlobPatterns(supportedFiles);
   for (const folder of dirs) {
-    const searcher = searchFiles(globPatterns, folder, symlinksEnabled, options.fileIgnores);
+    const searcher = searchFiles(globPatterns, folder, symlinksEnabled, fileIgnores);
     // eslint-disable-next-line no-await-in-loop
     for await (const filePath of searcher) {
-      const fileInfo = await getFileInfo(filePath.toString(), options.baseDir, false, cache);
+      const fileInfo = await getFileInfo(filePath.toString(), baseDir, false, cache);
       // dc ignore AttrAccessOnNull: false positive, there is a precondition with &&
-      if (fileInfo && fileInfo.size <= maxPayload) {
+      if (fileInfo && fileInfo.size <= maxPayload && !fileExcludes.includes(fileInfo.bundlePath)) {
```

Comment:
This would not work sadly. We've already tried to post-filter files based on exclusion rules, but it's too slow for too many customers. We need to inject the exclusion rules into the glob search in order to prune entire subtrees a-priori and not waste time scanning files one-by-one if we know they would be ignored anyway.
The whole distinction between `fileIgnores` and `fileExcludes` is irrelevant. All ignores and excludes at this level are in fact path exclusions, so please merge the two logics together (you will probably need to translate yaml exclusions to glob patterns).